internal_packages/keybase/lib/keybase-user.cjsx

```diff
@@ -0,0 +1,237 @@
+
+    if key?
+      PGPKeyStore.deleteKey(key)
+    else
+      console.error "Unable to fetch key for #{email}"
+
+  _addEmail: (email) =>
+    # associate another email address with this key
+    PGPKeyStore.addAddressToKey(@props.profile, email)
+
+  _addEmailInput: (contacts) =>
+    # when a new email is added to the list of emails
+    # flow is (click "add email") -> _addEmailClick -> (enter email) -> this
+    emails = _.pluck(contacts.to, 'email')
+    _.each(emails, @_addEmail)
+    @setState({inputEmail: false})
+
+  _addEmailClick: (event) =>
+    # create a text field in which to enter a new email to associate with a key
+    @setState({inputEmail: true})
+    # TODO focus on the new field
+    # React.findDOMNode(@refs.addNewEmail).focus()
+
+  _removeEmailClick: (event) =>
+    email = event.target.attributes['data-email'].value
+    @_removeEmail(email)
+
+  _removeEmail: (email) =>
+    PGPKeyStore.removeAddressFromKey(@props.profile, email)
+
+  render: =>
+    {profile} = @props
+
+    keybaseDetails = null
+    if profile.keybase_user?
+      keybase = profile.keybase_user
+      # username
+      username = keybase.components.username.val
+
+      # full name
+      if keybase.components.full_name?.val?
+        fullname = keybase.components.full_name.val
+      else
+        fullname = username
+
+      # profile picture
+      if keybase.thumbnail?
+        picture = keybase.thumbnail
+      else
+        picture = "#"
+        # TODO default picture
+
+      # various web accounts/profiles
+      possible_profiles = ["twitter", "github", "coinbase"]
+      profiles = _.map(possible_profiles, (possible) =>
+        if keybase.components[possible]?.val?
+          return "#{possible}: #{keybase.components[possible].val}"
+      )
+      profiles = _.reject(profiles, (profile) -> profile is undefined)
+
+      if profiles.length > 0
+        profiles =  _.map(profiles, (profile) ->
+          return <li>{ profile }</li>)
+        profileList = (<ul>{ profiles }</ul>)
+      else
+        profileList = null
+
+      keybaseDetails = (<div className="details">
+        <h2>
+          { fullname }&nbsp;
+          <small className="profile-username">
+            ({ username })
+          </small>
+        </h2>
+
+        { profileList }
+      </div>)
+
+    # button to save/delete key
+    if profile.addresses?.length > 0
+      # TODO what about the other email(s) that might be in the list?
+      saved = PGPKeyStore.pubKeys(profile.addresses[0])
+    else
+      # can't possibly have them saved if we have no email for them
+      saved = false
+    if not saved
+      saveDeleteButton = (<button title="Import" className="btn btn-toolbar" onClick={ @_importKey } ref="button">
+        Import Key
+      </button>
+      )
+    else
+      saveDeleteButton = (<button title="Delete" className="btn btn-toolbar btn-danger" data-email={ profile.addresses[0] } onClick={ @_delete } ref="button">
```

Comment:
This would actually be a nice place to use a bound function. You could say:

```onClick={ => @_onDeleteEmail(profile.addresses[0]) } ```

And not need to attach the email to the react element's dataset. We do that occasionally, but it's a bit hairy.